spec/support/authentication_helpers.rb

```diff
@@ -0,0 +1,37 @@
+module AuthenticationHelpers
+  def authenticate!(start_url:, expected_text:)
+    @@username ||= begin
+      print "SUNet ID: "
+      username = $stdin.gets
+      username.strip
+    end
+
+    @@password ||= begin
+      print "Password: "
+      password = $stdin.noecho(&:gets)
+      # So the user knows we're off the password prompt
+      puts
```

Comment:
I know it looks weird to have the puts here, but if it's the last line in the begin/end block, it interferes with the return value. And if it's outside the block, there's unnecessary newlines in the spec output.